app/src/main/java/co/nimblehq/ui/screens/MainNavigator.kt

```diff
@@ -0,0 +1,35 @@
+package co.nimblehq.ui.screens
+
+import androidx.fragment.app.Fragment
+import co.nimblehq.R
+import co.nimblehq.ui.base.BaseNavigator
+import co.nimblehq.ui.base.BaseNavigatorImpl
+import co.nimblehq.ui.base.NavigationEvent
+import co.nimblehq.ui.screens.home.HomeFragmentDirections.Companion.actionHomeFragmentToSecondFragment
+import co.nimblehq.ui.screens.second.SecondBundle
+import javax.inject.Inject
+
+interface MainNavigator : BaseNavigator
+
+class MainNavigatorImpl @Inject constructor(
+    fragment: Fragment
+) : BaseNavigatorImpl(fragment), MainNavigator {
+
+    override val navHostFragmentId = R.id.navHostFragment
+
+    override fun navigate(event: NavigationEvent) {
+        when (event) {
+            is NavigationEvent.Second -> navigateToSecond(event.bundle)
+        }
+    }
+
+    private fun navigateToSecond(bundle: SecondBundle) {
+        val navController = findNavController()
+        when (navController?.currentDestination?.id) {
+            R.id.homeFragment -> navController.navigate(
+                actionHomeFragmentToSecondFragment(bundle)
+            )
+            else -> unsupportedNavigation()
```

Comment:
if it currently has only 1 route from A -> B then I think we can omit this as well